keps/2724-topology-aware-scheduling/README.md

```diff
@@ -743,6 +748,133 @@ to ungate a Pod. The expectation is fulfilled if the Pod is observed as ungated
+at the moment of QuotaReservation (first pass).
+
+```golang
+type PodSetAssignment struct {
+  ...
+  // delayedTopologyRequest indicates the topology assignment is delayed.
+  // Topology assignment might be delayed in case there is ProvisioningRequest
+  // AdmissionCheck used.
+  // Kueue schedules the second pass of scheduling for each workload with at
+  // least one PodSet which has delayedTopologyRequest=true and without
+  // topologyAssignment.
+  //
+  // +optional
+  DelayedTopologyRequest *bool `json:"delayedTopologyRequest,omitempty"`
+}
+```
+
+This field will be particularly useful to implement functions such as
+[SyncAdmittedCondition](https://github.com/kubernetes-sigs/kueue/blob/f61f8e9549801e0f92c4f38fa06649cccfcc8d7d/pkg/workload/admissionchecks.go#L33)
+without the need to inspect the status of other objects (such as the
+ResourceFlavor and ClusterQueue in case of implicit defaulting).
+
+#### Targeting the newly provisioned nodes
+
+Some classes of provisioning requests may be always provisioning new nodes.
+In that case we need to give TAS the information about the node selector which
+should be used to target the newly provision nodes. We propose all the necessary
+information is contained in the ProvisioningRequest status, in the
+[provisioningClassDetails](https://github.com/kubernetes/autoscaler/blob/79a1375afe82416c528e72448ca5d5d96e6ad4ba/cluster-autoscaler/apis/provisioningrequest/autoscaling.x-k8s.io/v1/types.go#L169)
+field.
+
+In order to use the information we extend the API for ProvisioningRequestConfig
+as follows:
+
+```golang
+
+type ProvisioningRequestConfigSpec struct{
+...
+
+   // podSetUpdates specifies the update of the workload's PodSetUpdates which
+   // are used to target the provisioned nodes.
+   //
+   // +optional
+   PodSetUpdates []ProvisioningRequestPodSetUpdate `json:"podSetUpdates,omitempty"`
+}
+
+// ProvisioningRequestPodSetUpdateType specifies the type of the PodSetUpdate
+// +enum
+type ProvisioningRequestPodSetUpdateType string
+
+const (
+   // This type indicates this is PodSetUpdate for NodeSelector.
+   ProvisioningRequestPodSetUpdateTypeNodeSelector ProvisioningRequestPodSetUpdateType = "NodeSelector"
+
+
+   // This type indicates this is PodSetUpdate for Annotations.
+   ProvisioningRequestPodSetUpdateTypeAnnotations ProvisioningRequestPodSetUpdateType = "Annotations"
+
+
+   // This type indicates this is PodSetUpdate for Labels.
+   ProvisioningRequestPodSetUpdateTypeLabels ProvisioningRequestPodSetUpdateType = "Labels"
+)
+
+type ProvisioningRequestPodSetUpdate struct {
+   // podSetUpdates specifies the update of the workload's PodSetUpdates which
+   // are used to target the provisioned nodes.
+   //
+   // +required
+   // +kubebuilder:validation:Enum={NodeSelector,Annotations,Labels}
+   Type ProvisioningRequestPodSetUpdateType `json:"type,omitempty"`
+
+
+   // key specifies the key or the PodSetUpdate.
```

Comment:
Can we make it more clear this field indicates key of nodeSelector/Annotation/Label

Reply:
Yes, sure as consulted with @PBundyra on priv I will make it more explicit
```yaml
podSetUpdates:
  nodeSelector:
    key:
    ValueFromProvClassDetail
```
we don't have a use case for supporting annotations and labels as of now. Also, this new api will allow us to use kubebuilder for validation as key for labels, annotations and nodeSelectors may differ.

Reply:
updated, ptal